lib/facter/resolvers/linux/networking.rb

```diff
@@ -62,6 +64,30 @@ def physical(ifname)
                                                          end
           end
 
+          def duplex(interface_name)
+            # not all interfaces support this, wifi for example causes an EINVAL (Invalid argument)
+            unless interface_name.start_with?('lo')
+              begin
+                plex = Facter::Util::FileHelper.safe_read("/sys/class/net/#{interface_name}/duplex", nil)
+                @fact_list[:interfaces][interface_name][:duplex] = plex.strip if plex
+              rescue
+                @fact_list[:interfaces][interface_name][:duplex] = nil
```

Comment:
Could you log the error at debug level:

```suggestion
              rescue => e
                @log.debug("Failed to read '/sys/class/net/#{interface_name}/duplex': #{e.message}")
                @fact_list[:interfaces][interface_name][:duplex] = nil
              end
```